release/internal/config/operator.go

```diff
@@ -0,0 +1,60 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/projectcalico/calico/release/internal/command"
+	"github.com/projectcalico/calico/release/internal/version"
+)
+
+type OperatorConfig struct {
+	// GitRemote is the remote for the git repository
+	GitRemote string `envconfig:"OPERATOR_GIT_REMOTE" default:"origin"`
+
+	// Organization is the GitHub organization for the operator
+	Organization string `envconfig:"OPERATOR_GIT_ORGANIZATION" default:"tigera"`
+
+	// GitRepository is the repository for the operator
+	GitRepository string `envconfig:"OPERATOR_GIT_REPOSITORY" default:"operator"`
+
+	// Branch is the repository for the operator
+	Branch string `envconfig:"OPERATOR_BRANCH" default:"master"`
+
+	// RepoReleaseBranchPrefix is the prefix for the release branch
+	RepoReleaseBranchPrefix string `envconfig:"OPERATOR_RELEASE_BRANCH_PREFIX" default:"release"`
+
+	// DevTagSuffix is the suffix for the development tag
+	DevTagSuffix string `envconfig:"OPERATOR_DEV_TAG_SUFFIX" default:"0.dev"`
+
+	// Dir is the directory to clone the operator repository
+	Dir string `envconfig:"OPERATOR_DIR"`
+
+	// Image is the image for Tigera operator
+	Image string `envconfig:"OPERATOR_IMAGE" default:"tigera/operator"`
+
+	// Registry is the registry for Tigera operator
+	Registry string `envconfig:"OPERATOR_REGISTRY" default:"quay.io"`
```

Comment:
used by pinnedversion. Enterprise defines a different repo

Reply:
Ok, yeah I think we have a few types of configuration evolving here:

1. Things that vary between Enterprise and OSS Calico.
2. Things that devs may want to set for their own development.
3. Things that vary based on if they are run in CI or another environment.

For these, I personally think they are best handled with three different approaches

1. Using constants that are modified in each repository, avoiding merge-conflicts and not exposing values that don't change unnecessarily.
2. Via CLI arguments - these are self-documenting, don't require digging through the code.
3. Via env vars, since they vary based on where the code is being run.

Some config options fit more than one of these categories, and may need a bit more thought.

Reply:
@radTuti could you explain that a bit? My understanding is that the repository that we publish the operator to is always quay.io/tigera/ anyway?